subscriber/podaac_data_subscriber.py

```diff
@@ -221,24 +226,50 @@ def run():
     ## Sentinel-6 MF datasets also have *.bufr.bin, *.DBL, *.rnx, *.dat
     extensions = args.extensions
 
-    data = args.outputDirectory
-    #You should change `data` to a suitable download path on your file system.
-
-
-    bounding_extent=args.bbox
-    #Change this to whatever extent you need. Format is W Longitude,S Latitude,E Longitude,N Latitude
+    #Error catching for output directory specifications
+    #Must specify -d output path or one time-based output directory flag
+    if not args.outputDirectory and not any([args.cycle, args.dydoy, args.dymd]):
+        parser.error('No output directory specified, '
+                     'specify output directory with -d, '
+                     'or use one of the following flags: -dc, -dydoy, -dymd')
 
+    if args.outputDirectory and \
+            sum(map(bool, [args.cycle, args.dydoy, args.dymd])) == 1:
+        parser.error('Too many output directory options specified, '
+                     'Please specify -d or one flag '
+                     'from -dc, -dydoy, or -dymd')
 
+    if not args.outputDirectory and sum(map(bool, [args.cycle, args.dydoy, args.dymd])) != 1:
+        parser.error('Too many output directory flags specified, '
+                     'Please specify exactly one flag '
+                     'from -dc, -dydoy, or -dymd')
 
     # **The search retrieves granules ingested during the last `n` minutes.** A file in your local data dir  file that tracks updates to your data directory, if one file exists.
-
-
     timestamp = (datetime.utcnow()-timedelta(minutes=mins)).strftime("%Y-%m-%dT%H:%M:%SZ")
 
+    time_now = datetime.utcnow()
+    year = time_now.strftime('%Y')
+    month = time_now.strftime('%m')
+    day = time_now.strftime('%d')
+    day_of_year = time_now.strftime('%j')
```

Comment:
You're using the current date/time to create the directories before we get the data. We want the directories to be created based on the _start_time_ of the cmr start time for the granule.

The difference here is if you download a weeks worth of data on DOY=day0, you'd have /year/day0/files where files is any file downloaded on day0.

the expected behavior is to download data into the year/doy of the granule file start time (from CMR): year/day1/files ... /year/day2/files... /year/dayN/files

Reply:
Gotcha, thanks for the clarify that.  I have a changed this to use the data start_time rather than time of download.  The data start time will be sourced from `['umm']['TemporalExtent']['RangeDateTime']['BeginningDateTime']` in the CMR response.